lib/ecto/type.ex

```diff
@@ -1298,10 +1294,6 @@ defmodule Ecto.Type do
     {:ok, acc}
   end
 
-  defp map(_, _, _, _) do
-    :error
-  end
```

Comment:
I'm not sure removing this clause is correct. I believe it handles a case where we get something that's not a map or keyword from adapter